src/frequenz/sdk/actor/power_distributing/_component_managers/_ev_charger_manager/_ev_charger_manager.py

```diff
@@ -0,0 +1,448 @@
+from ..._component_pool_status_tracker import ComponentPoolStatusTracker
+
+# from .._component_pool_status_tracker import ComponentPoolStatusTracker
+from ..._component_status import ComponentPoolStatus, EVChargerStatusTracker
+from ...request import Request
+from ...result import PartialFailure, Result, Success
+from .._component_manager import ComponentManager
+from ._config import EVDistributionConfig
+from ._states import EvcState, EvcStates
+
+_logger = logging.getLogger(__name__)
+
+_DEFAULT_API_REQUEST_TIMEOUT = timedelta(seconds=5.0)
+_TGT_POWER_RESEND_INTERVAL = timedelta(seconds=5.0)
+
+
+class EVChargerManager(ComponentManager):
+    """Manage ev chargers for the power distributor."""
+
+    @override
+    def __init__(
+        self,
+        component_pool_status_sender: Sender[ComponentPoolStatus],
+        results_sender: Sender[Result],
+    ):
+        """Initialize the ev charger data manager.
+
+        Args:
+            component_pool_status_sender: Channel for sending information about which
+                components are expected to be working.
+            results_sender: Channel for sending results of power distribution.
+        """
+        self._results_sender = results_sender
+        self._ev_charger_ids = self._get_ev_charger_ids()
+        self._evc_states = EvcStates()
+        self._voltage_cache: LatestValueCache[Sample3Phase[Voltage]] = LatestValueCache(
+            microgrid.voltage().new_receiver()
+        )
+        self._config = EVDistributionConfig(component_ids=self._ev_charger_ids)
+        self._component_pool_status_tracker = ComponentPoolStatusTracker(
+            component_ids=self._ev_charger_ids,
+            component_status_sender=component_pool_status_sender,
+            max_data_age=timedelta(seconds=10.0),
+            max_blocking_duration=timedelta(seconds=30.0),
+            component_status_tracker_type=EVChargerStatusTracker,
+        )
+        self._target_power = Power.zero()
+        self._target_power_channel = Broadcast[Request](name="target_power")
+        self._target_power_tx = self._target_power_channel.new_sender()
+        self._task: asyncio.Task[None] | None = None
+        self._latest_request: Request = Request(Power.zero(), set())
+
+    @override
+    def component_ids(self) -> collections.abc.Set[int]:
+        """Return the set of ev charger ids."""
+        return self._ev_charger_ids
+
+    @override
+    async def start(self) -> None:
+        """Start the ev charger data manager."""
+        self._task = asyncio.create_task(self._run_forever())
+
+    @override
+    async def distribute_power(self, request: Request) -> None:
+        """Distribute the requested power to the ev chargers.
+
+        Args:
+            request: Request to get the distribution for.
+        """
+        await self._target_power_tx.send(request)
+
+    @override
+    async def stop(self) -> None:
+        """Stop the ev charger data manager."""
```

Comment:
There is no stopping the data manager?

Reply:
Oops, this is supposed to stop the component pool status tracker.  I've added a fixup commit.